### PR TITLE
client to server bandwidth measurement API

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
PTAL @vadmeste @klauspost

**Comment by harshavardhana** [comment]
NOTE: the measurement here is only for throughput so there is no RX/TX involved in this.

**Comment by klauspost** [comment]
Shouldn't we add the download benchmark right away? Do we have a speedtest endpoint we can pull data from?

**Comment by harshavardhana** [comment]
> Shouldn't we add the download benchmark right away? Do we have a speedtest endpoint we can pull data from?
> 
> In many cases bandwidth will be async.

There is no reason to do that you can saturate the link by simply sending data @klauspost - since its pure network doesn't need write/read like differentiation IMO.

**Comment by klauspost** [comment]
What do you mean? There are many async connection types out there. Measuring upload only gives half the picture.

**Comment by harshavardhana** [comment]
> What do you mean? There are many async connection types out there. Measuring upload only gives half the picture.

We are only validating what is the network throughput from client to server, this doesn't measure anything else. Of course, the server is not sending data back to the client here, but that is not useful IMO. 

Because all we care to show is the X bandwidth fully saturated value what is possible from client to server. In a full duplex connection such as 100g the 12GiB/sec value itself shows what is the overall bandwidth.

**Comment by harshavardhana** [comment]
any other questions @klauspost @vadmeste ?

File: perf-net.go [hunk 1]
```diff
@@ -56,3 +61,81 @@ func (adm *AdminClient) Netperf(ctx context.Context, duration time.Duration) (re
 	err = json.NewDecoder(resp.Body).Decode(&result)
 	return result, err
 }
+
+type NetperfClientResult struct {
+	Endpoint  string `json:"endpoint"`
+	Bandwidth uint64 `json:"bw"`
```
Comment (klauspost) [inline perf-net.go:67]:
```suggestion
	UploadBandwidth uint64 `json:"up_bps"`
```

**Review by klauspost — CHANGES_REQUESTED** [review]
> In a full duplex connection such as 100g, the 12GiB/sec value itself shows what is the overall bandwidth.

There are many async connections that do not provide close to the same upload/download speeds.

I don't mind that you don't want to add download test now. But then let's name the field what it is, so we can add download later.

**Comment by harshavardhana** [comment]
Will re-submit when we need this..

**Pull request closed without merging**